### ISSUE TITLE
Unable to Reproduce Results on Simple Distributions—Does EqM Require a Sparse (Dirac-Like) Target?

### ISSUE THREAD

**Issue by ischencheng** [raywang4/EqM#5]
I attempted to transport a Gaussian distribution to the two-moons distribution (scikit-learn), but I couldn’t get it to work. Am I misunderstanding the requirements? Does the target distribution need to be Dirac-like? In your paper, both the high-dimensional image examples and the 2D examples appear to be of this form. Here is my notebook.
https://colab.research.google.com/drive/11Ps9FZWqrJj2KnX0s7PJlBloud0DbYvC?usp=sharing

**Comment by raywang4** [comment]
Indeed, our derivations were based on the assumption that data is high dimensional, and it may not work as well on simple distributions like this one. Something you could try: I see that you directly adopted the $$c$$ function from our ImageNet experiments, which is fine but tuning it may give you better results.

**Comment by lukehouston85** [comment]
> I attempted to transport a Gaussian distribution to the two-moons distribution (scikit-learn), but I couldn’t get it to work. Am I misunderstanding the requirements? Does the target distribution need to be Dirac-like? In your paper, both the high-dimensional image examples and the 2D examples appear to be of this form. Here is my notebook. https://colab.research.google.com/drive/11Ps9FZWqrJj2KnX0s7PJlBloud0DbYvC?usp=sharing

Hi, I ran into the same issue when trying simple distributions like the two-moons case. In my opinion , the essential thing for training the EqM model is to make sure that margin gradient vanishes at real data points, so you need to manually add some points with t=1 when generating the training data to enforce that. I have modified the content of your notebook based on my understanding (also replaced the ct function with prescribed schedulers, which is just a different perspective of EqM), the results seem to better now. If you are still interested, please take a look at the link provided below.
https://drive.google.com/file/d/1hQEUmOKRE8lIGrcTKgicihoPbp_Csym6/view?usp=sharing
